src/swap/saga.ts

```diff
@@ -467,8 +468,36 @@ export function* swapSubmitPreparedSaga(action: PayloadAction<SwapInfoPrepared>)
 
     const feeCurrencyId = getTokenId(networkId, getFeeCurrency(preparedTransactions))
 
-    const swapTxHash = txHashes[txHashes.length - 1]
+    // if there is an approval transaction, add a standby transaction for it
+    if (preparedTransactions.length > 1) {
+      const approvalTx = preparedTransactions[0]
+      const { functionName, args } = yield* call(decodeFunctionData, {
+        abi: erc20.abi,
+        data: approvalTx.data ?? '0x0',
```

Comment:
yes that's true. i added this as a condition to the if statement so this block is skipped if there is no data. i think this is nicer, because technically we don't know if it's an approval tx until the tx is decoded so the `approvalTx` variable there is not named correctly. thanks!